bin/pin-dependency.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Istio Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+ISTIO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+# Usage:
+#   hack/pin-dependency.sh $MODULE $SHA-OR-TAG
```

Comment:
Multiple places where _hack_ has been changed to _bin_ in the directory structure, but not in the comments.